platform-hub-api/spec/controllers/contact_lists_controller_spec.rb

```diff
@@ -41,51 +91,106 @@
   end
 
   describe 'PUT #update' do
-    let :new_email_addresses do
+    let(:id) { 'foo' }
+    let :email_addresses do
       [ 'email1', 'email2' ]
     end
 
     let :put_params do
       {
-        id: @contact_list.id,
+        id: id,
         contact_list: {
-          email_addresses: new_email_addresses
+          email_addresses: email_addresses
         }
       }
     end
 
+    it_behaves_like 'unauthenticated not allowed'  do
+      before do
+        put :update, params: put_params
+      end
+    end
+
+    it_behaves_like 'authenticated' do
+
+      it_behaves_like 'not an admin so forbidden'  do
+        before do
+          put :update, params: put_params
+        end
+      end
+
+      it_behaves_like 'an admin' do
+        context 'for a contact list that doesn\'t yet exist' do
+          it 'should create a new contact list with the ID specified' do
+            expect(ContactList.count).to eq 0
+            expect(Audit.count).to eq 0
+            put :update, params: put_params
+            expect(response).to be_success
+            expect(ContactList.count).to eq 1
```

Comment:
I've avoided this style in the controller tests so far – rather I've used a structure like:

```
<assert preconditions>
<do thing>
<assert postconditions>
```

(not generally a fan of that block style since you can only express one change in that condition, unless you chain together lots of RSpec matchers, in which case it looks a bit messy :)